complex/src/lib.rs

```diff
@@ -740,6 +742,141 @@ impl<T> fmt::Binary for Complex<T> where
     }
 }
 
+impl<T> FromStr for Complex<T> where
+    T: FromStr + Num + PartialOrd + Clone
+{
+    type Err = ParseComplexError;
+
+    /// Parses `a +/- bi`; `ai +/- b`; `a`; or `bi` where `a` and `b` are of type `T`
+    fn from_str(s: &str) -> Result<Complex<T>, ParseComplexError>
+    {
+        let imag = match s.rfind('j') {
+            None => 'i',
+            _ => 'j'
+        };
+
+        let mut a = String::with_capacity(s.len());
+        let mut b = String::with_capacity(s.len());
```

Comment:
IIRC this is an artifact from my first attempt at a parser. We could probably end up reusing `s` for the real part and just allocate for the imaginary part. Is that what you're getting at?